api/v1alpha1/externaldns_webhook.go

```diff
@@ -100,6 +100,11 @@ func (r *ExternalDNS) validateFilters() error {
 }
 
 func (r *ExternalDNS) validateHostnameAnnotationPolicy() error {
+	if r.Spec.Source.Type == SourceTypeRoute {
```

Comment:
Should we reject the CR when source type is `Route` and the hostname annotation policy is specified?

Reply:
You mean when the hostname annotation policy is `Allow`? The route source [can use the hostname annotation](https://github.com/kubernetes-sigs/external-dns/blob/a3a9d4f27c1a02411585223059a5da5dcd837c31/source/source.go#L142).

Reply:
In that case when the policy is `Allow` we won't validate that the the `fqdnTemplate` is set altogether. Is that what we want?

Reply:
Well, yes, if the policy is `Allow`, the user can specify the custom hostname in the route object, so `FQDNTemplate` is not really needed.
But I believe it's still the valid option to have both. We already had a similar discussion [here](https://github.com/openshift/external-dns-operator/pull/26#discussion_r723310084).